src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/SeqGraph.java

```diff
@@ -269,12 +273,141 @@ protected SeqVertex mergeLinearChainVertex(final LinkedList<SeqVertex> linearCha
         return addedVertex;
     }
 
-    private static SeqVertex mergeLinearChainVertices(final Iterable<SeqVertex> vertices) {
+    /**
+     * Merges the given {@code vertices} into a single vertex.
+     * @param vertices {@link Iterable<SeqVertex>} of vertices to be merged together into a single vertex containing a linear sequence.
+     * @return A {@link SeqVertex} representing the data from all given {@code vertices} merged together.
+     */
+    protected SeqVertex mergeLinearChainVertices(final Iterable<SeqVertex> vertices) {
         final List<byte[]> seqs = new LinkedList<>();
         for ( final SeqVertex v : vertices ) {
             seqs.add(v.getSequence());
         }
         final byte[] seqsCat = Bytes.concat(seqs.toArray(new byte[][]{}));
         return new SeqVertex( seqsCat );
     }
+
+    /**
+     * Dumps this {@link SeqGraph} to disk as a GFA 1.0 file.
+     *
+     * GFA 1 spec is located here: http://gfa-spec.github.io/GFA-spec/GFA1.html
+     *
+     * @param baseName The base name of the output file.
+     */
+    public void serializeToGfa1Files(final String baseName) {
+        helpSerializeToGfaFile(baseName, false);
+    }
+
+    /**
+     * Dumps this {@link SeqGraph} to disk as a GFA 2.0 file.
+     *
+     * GFA 2 spec is located here: https://github.com/GFA-spec/GFA-spec/blob/master/GFA2.md
+     *
+     * @param baseName The base name of the output file.
+     */
+    public void serializeToGfa2Files(final String baseName) {
+        logger.warn("GFA2 Serialization is untested and may be incorrect.  USE AT YOUR OWN PERIL.");
+        helpSerializeToGfaFile(baseName, true);
+    }
+
+    private void helpSerializeToGfaFile(final String baseName, final boolean isGfa2) {
+
+        final String extension;
+        if ( isGfa2 ) {
+            extension = ".gfa2";
+        }
+        else {
+            extension = ".gfa";
+        }
+
+        try ( final PrintWriter writer = new PrintWriter(baseName + extension) ) {
```

Comment:
Assuming the gfa format doesn't enforce nodes  then links format it might be prudent to make this stream (or a writer with the same functionality) held by the tool so it can write the graph in chunks and forget about past graphs (eg cutting at contig levels). I think refacotring the writer this way would probably be prudent anyway even if you don't implement this now.